test/collections_test.rb

```diff
@@ -32,38 +32,57 @@
         assert_empty invalid_slugs, "Invalid item slugs #{invalid_slugs}"
       end
 
-      it "does not include items pointing to private or non-existent repos" do
-        invalid_repos = []
+      it "fails if a repository does not exist or is private" do
+        errors = []
 
         items_for_collection(collection).each do |item|
           next unless item.match?(USERNAME_AND_REPO_REGEX)
 
           url = URI("https://github.com/#{item}")
           http_status = Net::HTTP.get_response(url).code
 
-          unless http_status.match?(VALID_USER_AND_REPO_HTTP_STATUSES)
-            invalid_repos << item
+          unless %w[200 301].include?(http_status)
+            errors << "#{collection}: #{item} does not exist or is private"
           end
         end
 
-        assert_empty invalid_repos, "repositories #{invalid_repos} do not exist or are private"
+        assert_empty errors
       end
 
-      it "does not include items pointing to non-existent users or organizations" do
-        invalid_users = []
+      it "fails if a user or organization does not exist" do
```

Comment:
First let me say I think testing for the removal and renaming of collection items is a good thing. But I am worried this test could be flaky. I am not sure that will be a problem, but I wanted to mention it since we are making http requests. Are you concerned about that?

Reply:
@damaneice Totally – I've shared the same concerns. Haven't seen it flake so far though, which is pleasantly surprising 😄. We can all keep an eye out, and if it starts acting up look at solutions.